polynote-frontend/polynote/state/state_handler.ts

```diff
@@ -194,14 +196,16 @@ class ObserverDict<T> {
                 child.forEachAt(path, fn);
             }
         } else {
+            // NOTE: copy path so mutation doesn't affect upstream.
+            path = [...path];
```

Comment:
you could do this once in `forEach` rather than every time in `forEachAt`. As long as the path array that's passed into `forEach` doesn't itself get mutated.

Reply:
done!